File: app/src/main/java/com/softaai/designpatterns/java/creational/AbstractFactory/color/ColorFactory.java
package com.softaai.designpatterns.java.creational.AbstractFactory.color;

import com.softaai.designpatterns.java.creational.AbstractFactory.AbstractFactory;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.Circle;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.Rectangle;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.Shape;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.Square;

/**
 * Created by Amol Pawar on 27-03-2019.
 * softAai Apps
 */
public class ColorFactory extends AbstractFactory {

    public Color getColor(String color){
        if(color == null){
            return null;
        }
        if(color.equalsIgnoreCase("RED")){
            return new Red();
        }
        else if(color.equalsIgnoreCase("Blue")){
            return new Blue();
        }
        else if(color.equalsIgnoreCase("Green")){
            return new Green();
        }
        return null;
    }

    @Override
    public Shape getShape(String shape) {
        return null;
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/AbstractFactory/color/ColorFactory.kt
package com.softaai.designpatterns.kotlin.creational.AbstractFactory.color

import com.softaai.designpatterns.kotlin.creational.AbstractFactory.AbstractFactory
import com.softaai.designpatterns.kotlin.creational.AbstractFactory.shape.Shape


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
class ColorFactory : AbstractFactory() {
    override fun getColor(color: String): Color? {
        if(color.equals("RED", ignoreCase = true)){
            return Red()
        }
        else if(color.equals("BLUE", ignoreCase = true)){
            return Blue()
        }
        else if(color.equals("GREEN", ignoreCase = true)){
            return Green()
        }
        return null
    }

    override fun getShape(shape: String): Shape? {
        return null
    }
}